scripts/library.sh

```diff
@@ -487,9 +490,34 @@ function get_knative_base_yaml_source() {
   if [[ ${branch_name} =~ ^release-[0-9\.]+$ ]]; then
     # Get the latest tag name for the current branch, which is likely formatted as v0.5.0
     local tag_name="$(git describe --tags --abbrev=0)"
-    knative_base_yaml_source="https://storage.googleapis.com/knative-releases/@/previous/${tag_name}"
+    # The other repo might not have this tag, so we need to find the first available tag before it.
+    echo "$(find_latest_release_yaml_source ${tag_name} ${repo_name} ${source_name})"
+  else
+    local knative_base_yaml_source="https://storage.googleapis.com/knative-nightly/@/latest"
+    echo "${knative_base_yaml_source/@/${repo_name}}/${source_name}.yaml"
```

Comment:
The string replacement is not useful anymore, making this hard to read. Just assemble the final string, here.